tests/s3/versioning/test_delete_object_tagging.py

```diff
@@ -0,0 +1,164 @@
+        create_file(fpath=self.file_path, count=1)
+        LOGGER.info("Created file: %s", self.file_path)
+        self.bucket_name = f"tag-bkt-{time.perf_counter_ns()}"
+        self.object_name = f"tag-obj-{time.perf_counter_ns()}"
+        res = self.s3_test_obj.create_bucket(self.bucket_name)
+        assert_utils.assert_true(res[0], res[1])
+        assert_utils.assert_equal(res[1], self.bucket_name, res[1])
+        LOGGER.info("Created a bucket with name : %s", self.bucket_name)
+
+    def teardown_method(self):
+        """
+        Function will be perform cleanup after each test case.
+        """
+        LOGGER.info("STARTED: Teardown operations")
+        LOGGER.info("Clean : %s", self.test_dir_path)
+        if path_exists(self.file_path):
+            res = remove_file(self.file_path)
+            LOGGER.info("cleaned path: %s, res: %s", self.file_path, res)
+        if path_exists(self.test_dir_path):
+            remove_dirs(self.test_dir_path)
+        LOGGER.info("Cleanup test directory: %s", self.test_dir_path)
+        # DELETE Object with VersionId is WIP, uncomment once feature is available
+        # res = self.s3_test_obj.bucket_list()
+        # pref_list = []
+        # for bucket_name in res[1]:
+        #     if bucket_name.startswith("tag-bkt"):
+        #         empty_versioned_bucket(self.s3_ver_obj, bucket_name)
+        #         pref_list.append(bucket_name)
+        # if pref_list:
+        #     res = self.s3_test_obj.delete_multiple_buckets(pref_list)
+        #     assert_utils.assert_true(res[0], res[1])
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40435")
+    @CTFailOn(error_handler)
+    def test_del_obj_tag_40435(self):
+        """
+        Test DELETE object tagging for pre-existing object in a versioning enabled bucket
+        """
+        LOGGER.info("STARTED: Test DELETE object tagging for pre-existing object in a versioning "
+                    "enabled bucket")
+        ver_tag = dict()
+        ver_tag.update({self.object_name: dict()})
+        versions = dict()
+        LOGGER.info("Step 1: Upload object %s before enabling versioning on bucket %s",
+                    self.object_name, self.bucket_name)
+        upload_version(self.s3_test_obj, bucket_name=self.bucket_name, file_path=self.file_path,
+                       object_name=self.object_name, versions_dict=versions, is_unversioned=True)
+        latest_v = versions[self.object_name]["version_history"][-1]
+        LOGGER.info("Step 1: Successfully uploaded object %s before enabling versioning on  "
+                    "bucket %s with version ID %s", self.object_name, self.bucket_name, latest_v)
+        LOGGER.info("Step 2: Perform PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+        resp = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
```

Comment:
resp and resp[0] will always be different. What is the use of comparison.
resp will be a tuple/list and repo[0] will be item from that?

Reply:
Assert if resp[0] is not true, and if otherwise, it will just print resp as message of failures
this is not getting compared. the main condition is just resp[0] should be true